integration-tests-ex/cases/src/test/java/org/apache/druid/testsEx/query/ITQueryErrorTest.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.testsEx.query;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.inject.Inject;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.druid.java.util.common.StringUtils;
+import org.apache.druid.query.QueryContexts;
+import org.apache.druid.server.coordination.ServerManagerForQueryErrorTest;
+import org.apache.druid.testing.utils.DataLoaderHelper;
+import org.apache.druid.testing.utils.SqlTestQueryHelper;
+import org.apache.druid.testing.utils.TestQueryHelper;
+import org.apache.druid.testsEx.categories.QueryError;
+import org.apache.druid.testsEx.config.BaseJUnitRule;
+import org.apache.druid.testsEx.indexer.AbstractIndexerTest;
+import org.apache.druid.testsEx.config.DruidTestRunner;
+import org.assertj.core.api.Assertions;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+/**
+ * This class tests various query failures.
```

Comment:
Each test category requires we start a cluster, run the test, and stop the cluster. At present, since we don't yet cache the Druid image, we also build the test image for every test. Given this, I wonder if we should combine the three query tests into a single category?

We can do that by having three test files that are in the same category. This works as long as the tests are independent of one another. That is, test A does not create items on the Druid server that will throw off test B.

If we do want to combine tests, it might make sense to first get them into GitHub as individual tests (to reduce risk), then follow up with another PR that does the combining.

Reply:
We could discuss about combining queryError and QueryRetry. But definition of historical for the Query test group is different that these two.